src/TotkRstbGenerator.Core/RstbGenerator.cs

```diff
@@ -18,19 +18,21 @@ public class RstbGenerator
     private readonly string _output;
     private readonly uint _padding;
 
-    public RstbGenerator(string romfs, string? output = null, uint padding = 0)
-    {
-        _romfs = romfs;
-        _output = output ?? romfs.GetRsizetableFile();
-        _padding = padding;
+    private static Object _lock = new Object();
 
-        string path = TotkConfig.Shared.RsizetablePath;
+    public RstbGenerator(string romfs, string? rstb = null, string? output = null, uint padding = 0)
+    {
+        string path = rstb ?? TotkConfig.Shared.RsizetablePath;
```

Comment:
The variable name can be more descriptive.

Also, a second check should be done if `sourceRstbPath` is not null to provide an error that makes sense. "The vanilla RSTB file could not be found ..." is no longer a valid error if `path` isn't always the vanilla file.

```suggestion
    public RstbGenerator(string romfs, string? sourceRstbPath = null, string? output = null, uint padding = 0)
    {
        // Throw if sourceRstbPath is not null and does not exist.
        // The path error is confusing when it may not be the vanilla file
        string path = sourceRstbPath ?? TotkConfig.Shared.RsizetablePath;
```